src/set.rs

```diff
@@ -0,0 +1,1114 @@
+//! A hash set implemented as an `OrderMap` where the value is `()`.
+
+use std::cmp::Ordering;
+use std::collections::hash_map::RandomState;
+use std::fmt;
+use std::iter::{FromIterator, Chain};
+use std::hash::{Hash, BuildHasher};
+use std::mem::replace;
+use std::ops::RangeFull;
+use std::ops::{BitAnd, BitOr, BitXor, Sub};
+
+use super::{OrderMap, Equivalent};
+
+/// A hash set where the iteration order of the values is independent of their
+/// hash values.
+///
+/// The interface is closely compatible with the standard `HashSet`, but also
+/// has additional features.
+///
+/// # Order
+///
+/// The values have a consistent order that is determined by the sequence of
+/// insertion and removal calls on the set. The order does not depend on the
+/// values or the hash function at all.
+///
+/// All iterators traverse the set in *the order*.
```

Comment:
We need a comment on the order in the set operation iterators

Reply:
Done.